I'm fixing the flaky test in another PR.

```
panic: test timed out after 10m0s
	running tests:
		TestStreamSuite (10m0s)
		TestStreamSuite/TestOneChangeIsBlockedByFile (9m48s)

goroutine 4517 [running]:
testing.(*M).startAlarm.func1()
	/usr/local/go/src/testing/testing.go:2802 +0x605
created by time.goFunc
	/usr/local/go/src/time/sleep.go:215 +0x45

goroutine 1 [chan receive, 10 minutes]:
testing.(*T).Run(0xc000402008, {0x123c8d0, 0xf}, 0x129d2a8)
	/usr/local/go/src/testing/testing.go:2109 +0xb3e
testing.runTests.func1(0xc000402008)
	/usr/local/go/src/testing/testing.go:2585 +0x86
testing.tRunner(0xc000402008, 0xc00038fad0)
	/usr/local/go/src/testing/testing.go:2036 +0x21d
testing.runTests({0x12416db, 0x14}, {0x125d23f, 0x31}, 0xc00011e558, {0x1b0a2a0, 0x1, 0x1}, {0xc27206093271524b, 0x8bb3c52e03, ...})
	/usr/local/go/src/testing/testing.go:2583 +0x9ea
testing.(*M).Run(0xc000146140)
	/usr/local/go/src/testing/testing.go:2443 +0xf4c
main.main()
	_testmain.go:46 +0x165

goroutine 18 [chan receive, 9 minutes]:
testing.(*T).Run(0xc000402248, {0x10b62be, 0x1c}, 0xc00011ee58)
	/usr/local/go/src/testing/testing.go:2109 +0xb3e
github.com/juju/tc.(*suiteRunner).run(0xc000132410, 0xc000402248)
	/home/jenkins/go/pkg/mod/github.com/juju/tc@v0.0.0-20251023013639-77c6a1d20e5a/check.go:183 +0x371
github.com/juju/tc.Run(0xc000402248, {0x122d3a0, 0xc0001700e0})
	/home/jenkins/go/pkg/mod/github.com/juju/tc@v0.0.0-20251023013639-77c6a1d20e5a/run.go:90 +0x4f
github.com/juju/juju/internal/changestream/stream.TestStreamSuite(0xc000402248)
	/home/jenkins/go/src/github.com/juju/juju/internal/changestream/stream/stream_test.go:43 +0xf8
testing.tRunner(0xc000402248, 0x129d2a8)
	/usr/local/go/src/testing/testing.go:2036 +0x21d
created by testing.(*T).Run in goroutine 1
	/usr/local/go/src/testing/testing.go:2101 +0xb13

goroutine 4504 [chan send, 9 minutes]:
github.com/juju/juju/internal/changestream/stream.(*streamSuite).TestOneChangeIsBlockedByFile.func1.1()
	/home/jenkins/go/src/github.com/juju/juju/internal/changestream/stream/stream_test.go:851 +0x92
created by github.com/juju/juju/internal/changestream/stream.(*streamSuite).TestOneChangeIsBlockedByFile.func1 in goroutine 4160
	/home/jenkins/go/src/github.com/juju/juju/internal/changestream/stream/stream_test.go:849 +0x129

goroutine 4160 [select, 9 minutes]:
github.com/juju/worker/v5/workertest.CheckKilled({0x12b2f20, 0xc0005ac640}, {0x12aa9c8, 0xc000000000})
	/home/jenkins/go/pkg/mod/github.com/juju/worker/v5@v5.0.0/workertest/check.go:68 +0x25a
github.com/juju/worker/v5/workertest.CheckKill({0x12b2f20, 0xc0005ac640}, {0x12aa9c8, 0xc000000000})
	/home/jenkins/go/pkg/mod/github.com/juju/worker/v5@v5.0.0/workertest/check.go:86 +0x65
github.com/juju/worker/v5/workertest.DirtyKill({0x12b2f20, 0xc0005ac640}, {0x12aa9c8, 0xc000000000})
	/home/jenkins/go/pkg/mod/github.com/juju/worker/v5@v5.0.0/workertest/check.go:117 +0x5b
runtime.Goexit()
	/usr/local/go/src/runtime/panic.go:694 +0x5e
testing.(*common).FailNow(0xc000272008)
	/usr/local/go/src/testing/testing.go:1022 +0x7b
testing.(*common).Fatal(0xc000272008, {0xc00005b0f8, 0x1, 0x1})
	/usr/local/go/src/testing/testing.go:1221 +0x85
github.com/juju/juju/internal/changestream/stream.(*streamSuite).TestOneChangeIsBlockedByFile.func1(0x1)
	/home/jenkins/go/src/github.com/juju/juju/internal/changestream/stream/stream_test.go:856 +0x20a
github.com/juju/juju/internal/changestream/stream.(*streamSuite).TestOneChangeIsBlockedByFile(0xc0001700e0, 0xc0005ac640)
	/home/jenkins/go/src/github.com/juju/juju/internal/changestream/stream/stream_test.go:860 +0x48f
reflect.Value.call({0x122d3a0?, 0xc0001700e0?, 0xc00005bb60?}, {0x12326f3, 0x4}, {0xc00005bcc8, 0x1, 0x55309e?})
	/usr/local/go/src/reflect/value.go:586 +0x1225
reflect.Value.Call({0x122d3a0?, 0xc0001700e0?, 0x0?}, {0xc00005bcc8, 0x1, 0x1})
	/usr/local/go/src/reflect/value.go:369 +0xb6
github.com/juju/tc.(*methodType).Call(0xc000117340, 0xc0005ac640)
	/home/jenkins/go/pkg/mod/github.com/juju/tc@v0.0.0-20251023013639-77c6a1d20e5a/check.go:88 +0x2af
github.com/juju/tc.(*suiteRunner).runTest(0xc000132410, 0xc000272008, 0xc000117340)
	/home/jenkins/go/pkg/mod/github.com/juju/tc@v0.0.0-20251023013639-77c6a1d20e5a/check.go:225 +0x626
github.com/juju/tc.(*suiteRunner).run.func3(0xc000272008)
	/home/jenkins/go/pkg/mod/github.com/juju/tc@v0.0.0-20251023013639-77c6a1d20e5a/check.go:184 +0x45
testing.tRunner(0xc000272008, 0xc00011ee58)
	/usr/local/go/src/testing/testing.go:2036 +0x21d
created by testing.(*T).Run in goroutine 18
	/usr/local/go/src/testing/testing.go:2101 +0xb13

goroutine 4161 [select, 9 minutes]:
database/sql.(*DB).connectionOpener(0xc0003420d0, {0x12afa18, 0xc000133950})
	/usr/local/go/src/database/sql/sql.go:1261 +0xed
created by database/sql.OpenDB in goroutine 4160
	/usr/local/go/src/database/sql/sql.go:841 +0x287

goroutine 4516 [chan receive, 9 minutes]:
gopkg.in/tomb%2ev2.(*Tomb).Wait(0xc000000008)
	/home/jenkins/go/pkg/mod/gopkg.in/tomb.v2@v2.0.0-20161208151619-d5d1b5820637/tomb.go:126 +0x4f
github.com/juju/juju/internal/changestream/stream.(*Stream).Wait(0xc000000000)
	/home/jenkins/go/src/github.com/juju/juju/internal/changestream/stream/stream.go:220 +0x2d
github.com/juju/worker/v5/workertest.CheckKilled.func1()
	/home/jenkins/go/pkg/mod/github.com/juju/worker/v5@v5.0.0/workertest/check.go:64 +0x43
created by github.com/juju/worker/v5/workertest.CheckKilled in goroutine 4160
	/home/jenkins/go/pkg/mod/github.com/juju/worker/v5@v5.0.0/workertest/check.go:63 +0x131
```

--- a/domain/crossmodelrelation/state/model/offer.go
+++ b/domain/crossmodelrelation/state/model/offer.go
@@ -93,13 +93,11 @@ func (st *State) ValidateApplicationAndEndpointsForOffer(
 	// types. scope_id is container scope, which is not allowed to be offered.
 	stmt, err := st.Prepare(`
 SELECT COUNT(*) AS &countResult.count
-FROM   application AS a
-JOIN   application_endpoint AS ae ON a.uuid = ae.application_uuid
+FROM   application_endpoint AS ae
 JOIN   charm_relation AS cr ON ae.charm_relation_uuid = cr.uuid
-WHERE  a.uuid = $uuid.uuid
 AND    ae.uuid IN ($uuids[:])
 AND    cr.scope_id == 1
-`, uuids{}, uuid{}, countResult{})
+`, uuids{}, countResult{})
 	if err != nil {
 		return "", errors.Errorf("preparing application and endpoint validation query: %w", err)
 	}
@@ -123,7 +121,7 @@ AND    cr.scope_id == 1
 			return errors.Capture(err)
 		}
 
-		err = tx.Query(ctx, stmt, uuid{UUID: applicationUUID}, uuids(endpointUUIDs)).Get(&count)
+		err = tx.Query(ctx, stmt, uuids(endpointUUIDs)).Get(&count)
 		if errors.Is(err, sqlair.ErrNoRows) {
 			// No endpoints with container types, validation successful.
 			return nil

